### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by lazovicff**

### CODE REVIEW

**Comment by lazovicff** [comment]
@astralblue @hypnagonia I need your approval, before merging

File: init.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+if [[ $(cargo --version) ]]; then
+    echo "Found cargo"
+else
+    curl https://sh.rustup.rs -sSf | sh -s -- -y
```
Comment (astralblue) [inline init.sh]: * The `curl` options (`-sSf`) should precede the positional argument (URL)
* If `curl` fails, `sh` executes empty stdin and succeeds; `sh` should fail and cause `init.sh` to exit with error.

```suggestion
    (curl -sSf https://sh.rustup.rs || echo "exit 1") | sh -s -- -y || exit
```

This way, if `curl` fails:

* The parenthesized group outputs `exit 1`;
* The `sh` then interprets this and exits with status 1;
* The `|| exit` at the end examines the shell's exit status, and if nonzero, aborts `init.sh` with the same exit status:
  * Upon `curl` success, the exit status is from the installation script itself;
  * Upon `curl` failure, the exit status is 1.

File: init.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    source $HOME/.cargo/env
```
Comment (astralblue) [inline init.sh]: Same, if the `env` script fails, `init.sh` should abort:
```suggestion
    source $HOME/.cargo/env || exit
```

File: rust-toolchain.toml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+[toolchain]
+channel = "1.70.0"
+components = ["rustfmt", "clippy", "rls", "rust-analysis"]
```
Comment (astralblue) [inline rust-toolchain.toml]: Just learning Rust so I may be missing some history/BCP here, but [the `rustup` book says `rls` is deprecated and replaced by `rust-analyzer`](https://rust-lang.github.io/rustup/concepts/components.html#components); do we still need the old RLS?

File: rust-toolchain.toml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+[toolchain]
+channel = "1.70.0"
```
Comment (astralblue) [inline rust-toolchain.toml]: 1.70.0 is affected by [CVE-2023-38497](https://groups.google.com/g/rustlang-security-announcements/c/PEZQleQ6IUY).

```suggestion
channel = "1.71.1"
```

**Review by astralblue — CHANGES_REQUESTED** [review]

File: rust-toolchain.toml (hunk 3 above)
Comment (lazovicff) [reply rust-toolchain.toml]: True, it's deprecated and we don't even use it.

**Comment by lazovicff** [comment]
@astralblue Can you re-review?

**Review by astralblue — APPROVED** [review]
LGTM, thanks!

**Pull request merged**